pulsar-client/src/main/java/com/yahoo/pulsar/client/impl/ConsumerBase.java

```diff
@@ -72,9 +72,14 @@ protected ConsumerBase(PulsarClientImpl client, String topic, String subscriptio
         }
         this.listenerExecutor = listenerExecutor;
         this.pendingReceives = Queues.newConcurrentLinkedQueue();
-        if (conf.getAckTimeoutMillis() != 0) {
+
+        // initialize unAckedMessageTracker if stats (client-ack-stats uses it to track ack-message for cumulative
+        // ack) or AckTimeOut-Task enabled 
+        if (client.getConfiguration().getStatsIntervalSeconds() > 0 || conf.getAckTimeoutMillis() != 0) {
```

Comment:
yes. 
But to know total number for ack messages for cumulative ack, we need similar thing as ```unAckedMessageTracker``` and  consumer do initialize ```unAckedMessageTracker``` if ackTimeout is configured. 
So, we can have two options:
- Only initialize ```unAckedMessageTracker``` if ackTimeOut is configured, else client will not get correct stats for ack-messages in case of cumulative ack.
- we can given separate flag/configuration (default: false) which client can explicitly set in case: ackTimeOut is disabled and needs correct stats for cumulative ack.